packages/migrate/types/NodePath.ts

```diff
@@ -0,0 +1,89 @@
+export interface INode extends Object {
+	id?: {
+		name: string;
+	};
+	arguments?: INode[];
+	body?: INode[];
+	elements?: INode[];
+	expression?: {
+		left: {
+			computed: boolean,
+			object: INode,
+			property: INode,
+			type: string,
+		},
+		operator: string,
+		right: INode,
+		type: string,
+	};
+	filter?: (p: (p: INode) => boolean) => INode[] | any;
+	find?: (objectExpression: object, filterExpression?: object) => INode;
+	forEach?: (p: (p: INode) => void) => void;
+	get?: (property: string) => INode;
+	remove?: (_?: void) => void;
+	nodes?: (_?: void) => INode[];
+	pop?: (_?: void) => INode;
+	key?: {
+		name: string;
+		value: INode | string | any;
```

Comment:
if the key resolves to any there's not much need of the interface

Reply:
I know. If we're keeping `INode | string`, then we'll have to do `typeof` check everywhere in the codebase. There are more than 50 places (like below) where we would have to do these checks. :disappointed: 

![image](https://user-images.githubusercontent.com/5961873/42633871-532796d4-85ff-11e8-8e71-2e31fad05407.png)